nss-day-cohort-29/nutshell-midnight-dream
# News Articles update and Logout functionality

# Description
News Articles screen now displays both current user's and each friend's articles, and they are displayed in order, by timestamp. Also added functionality to the "Logout" link, which clears the session.

## Type of change

# Testing Instructions for Change Made
1. `git fetch --all`
2. `git checkout news-friend-fetch`
3. Run grunt
4. Go to localhost and log in with `userName2` and `userEmail2@gmail.com`
5. On the "News" screen, you should see a mix of articles titled "User X Article" with different users (which shows that articles from different users are being displayed).
6. You should see "first," "second," "third," etc. in the article titles - make sure they are displaying in order (this shows that the timestamp is being ordered correctly).
7. Click the "Logout" link - it should take you to the login screen.
8. Now log in with `userName3` and `userEmail3@gmail.com`
9. Now you should see only one article, tied to that user (user3), since they have no current friends.

If my data isn't pushed up, insert these into your `database.json`:

```
"users": [
    {
      "id": 1,
      "name": "userName",
      "email": "userEmail@gmail.com"
    },
    {
      "id": 2,
      "name": "userName2",
      "email": "userEmail2@gmail.com"
    },
    {
      "name": "userName3",
      "email": "userEmail3@gmail.com",
      "id": 3
    },
    {
      "name": "userName4",
      "email": "userEmail4@gmail.com",
      "id": 4
    }
  ]
```

```
"articles": [
    {
      "id": 1,
      "title": "User 1 Article (fourth)",
      "url": "http://google.com",
      "synopsis": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus efficitur purus eget est rutrum tincidunt. Vestibulum a velit accumsan quam scelerisque efficitur. Proin tincidunt magna id orci ultricies vulputate. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Duis laoreet lacus nibh, id tincidunt enim euismod in. Sed egestas sapien orci, in vestibulum ipsum tempor id. Aenean eleifend mi dui, in consequat lacus laoreet quis.",
      "timestamp": "1457395221082",
      "userId": 1
    },
    {
      "id": 2,
      "title": "User 2 Article (second)",
      "url": "http://google.com",
      "synopsis": "Fusce turpis mi, sollicitudin quis maximus ac, facilisis vel lectus. Duis laoreet sollicitudin odio at viverra. Phasellus scelerisque, ante id lacinia cursus, elit nunc ullamcorper ipsum, vel faucibus sapien lectus nec nisl. In dui lorem, dignissim sit amet turpis pulvinar, congue malesuada ligula. Morbi dignissim, est vel lacinia mollis, magna ligula posuere arcu, rhoncus ultricies est dui vel risus.",
      "timestamp": "1247395221082",
      "userId": 2
    },
    {
      "id": 3,
      "title": "User 2 Article2 (third)",
      "url": "http://google.com",
      "synopsis": "Fusce turpis mi, sollicitudin quis maximus ac, facilisis vel lectus. Duis laoreet sollicitudin odio at viverra. Phasellus scelerisque, ante id lacinia cursus, elit nunc ullamcorper ipsum, vel faucibus sapien lectus nec nisl. In dui lorem, dignissim sit amet turpis pulvinar, congue malesuada ligula. Morbi dignissim, est vel lacinia mollis, magna ligula posuere arcu, rhoncus ultricies est dui vel risus.",
      "timestamp": "1447395221082",
      "userId": 2
    },
    {
      "id": 4,
      "title": "User 3 Article (first)",
      "url": "http://google.com",
      "synopsis": "Fusce turpis mi, sollicitudin quis maximus ac, facilisis vel lectus. Duis laoreet sollicitudin odio at viverra. Phasellus scelerisque, ante id lacinia cursus, elit nunc ullamcorper ipsum, vel faucibus sapien lectus nec nisl. In dui lorem, dignissim sit amet turpis pulvinar, congue malesuada ligula. Morbi dignissim, est vel lacinia mollis, magna ligula posuere arcu, rhoncus ultricies est dui vel risus.",
      "timestamp": "1047395221082",
      "userId": 3
    },
    {
      "id": 5,
      "title": "User 4 Article",
      "url": "http://google.com",
      "synopsis": "Fusce turpis mi, sollicitudin quis maximus ac, facilisis vel lectus. Duis laoreet sollicitudin odio at viverra. Phasellus scelerisque, ante id lacinia cursus, elit nunc ullamcorper ipsum, vel faucibus sapien lectus nec nisl. In dui lorem, dignissim sit amet turpis pulvinar, congue malesuada ligula. Morbi dignissim, est vel lacinia mollis, magna ligula posuere arcu, rhoncus ultricies est dui vel risus.",
      "timestamp": "1647495221082",
      "userId": 4
    },
    {
      "title": "timestamp test (fifth)",
      "synopsis": "time!",
      "url": "meg.com",
      "timestamp": 1547395221082,
      "date": "01/13/2019 at 10:0",
      "userId": 2,
      "id": 6
    },
    {
      "title": "another timestamp (sixth)",
      "synopsis": "time time time",
      "url": "meg.com",
      "timestamp": 1547395320738,
      "date": "01/13/2019 at 10:2",
      "userId": 2,
      "id": 7
    },
    {
      "title": "it's a new hour",
      "synopsis": "woo!",
      "url": "meg.com",
      "timestamp": 1547402892179,
      "date": "01/13/2019 at 12:08",
      "userId": "2",
      "id": 10
    }
  ]
```

```
"connections": [
    {
      "id": 1,
      "userId": 1,
      "otherId": 2
    },
    {
      "id": 2,
      "userId": 1,
      "otherId": 3
    },
    {
      "id": 3,
      "userId": 2,
      "otherId": 1
    },
    {
      "id": 4,
      "userId": 2,
      "otherId": 3
    }
  ]
```

# Checklist:
